server/management/commands/merge_accounts.py

```diff
@@ -0,0 +1,250 @@
+        )
+    user_data["transactions"] = transactions if player_id else 0
+
+    total_fields = len(user_data)
+    completed_fields = sum(bool(value) for key, value in user_data.items())
+    completeness_score = (completed_fields / total_fields) * 100
+    return completeness_score, user_data
+
+
+def compare_users(usernames: list[str]) -> dict[str, Any]:
+    data = {}
+    for username in usernames:
+        completeness_score, info = calculate_completeness_score(username)
+        data[username] = dict(score=completeness_score, **info)
+
+    pt = PrettyTable()
+    pt.field_names = ["User Data", *usernames]
+    user_fields = {
+        "WFDF ID": "player_profile__accreditation__wfdf_id",
+        "Vaccination": "player_profile__vaccination__name",
+        "Membership": None,
+        "UC Profile": "player_profile__ultimate_central_id",
+        "Registrations": "registrations",
+        "Transactions": "transactions",
+        "Player Profile": "player_profile",
+        "Completeness": "score",
+    }
+    pt.align["User Data"] = "l"
+    for label, attr in user_fields.items():
+        if label == "Membership":
+            start = "player_profile__membership__start_date"
+            end = "player_profile__membership__end_date"
+            active = "player_profile__membership__is_active"
+            row_info = [
+                f"{data[u][start]}-{data[u][end]}" if data[u][active] else "--" for u in usernames
+            ]
+        else:
+            row_info = [data[u].get(attr, "--") for u in usernames if attr]
+        divider = label == "Player Profile"
+        pt.add_row([label, *row_info], divider=divider)
+    print(pt)
+    return data
+
+
+def find_duplicate_email_usernames() -> list[list[str]]:
+    users = dict(User.objects.values_list("username", "email"))
+    emails_to_usernames: dict[str, list[str]] = {}
+    for username, email in users.items():
+        email_ = email.lower().strip()
+        if email_ in emails_to_usernames:
+            emails_to_usernames[email_].append(username)
+        else:
+            emails_to_usernames[email_] = [username]
+
+    return [usernames for usernames in emails_to_usernames.values() if len(usernames) > 1]
+
+
+class Command(BaseCommand):
+    help = "Command to merge accounts"
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument("usernames", nargs="*", help="Usernames of accounts to merge.")
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        usernames = options["usernames"]
+        if len(usernames) == 2:  # noqa: PLR2004
+            self.merge_users(usernames)
+        elif len(usernames) == 0:
+            duplicates = find_duplicate_email_usernames()
+            for usernames in duplicates:
+                self.stdout.write(self.style.NOTICE(f"Merging usernames: {','.join(usernames)}"))
+                self.merge_users(usernames)
```

Comment:
If we have more than 2 duplicate emails this will fail right? Should we handle that by considering 2 at a time in a loop till we merge all accounts?

Reply:
Yes, there's already a check that exits early and asks user to manually merge the accounts, 2 at a time.